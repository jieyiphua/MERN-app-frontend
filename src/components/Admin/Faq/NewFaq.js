import React from 'react';
import { Redirect, Link, withRouter } from 'react-router-dom';
import axios from 'axios';

class NewFaq extends React.Component {
  state = {
    question: "",
    answer: "",
    errors: []
  }

  onInputChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  onFormSubmit = async (event) => {
    event.preventDefault();

    const newFaq  = { 
      question: this.state.question,
      answer: this.state.answer
    };

    try {
      const response = await axios.post('https://deployment-mern-backend-tessivanjayz.gardtess.now.sh/faq/new', newFaq);
    } catch (error) {
      this.setState({
        errors: error.response
      });
    }
    this.props.history.push('/faqs')
  }

  render() {
    const { authentication } = this.props

    if (!authentication) {
      return <Redirect to="/admin/login" />;
    } 
    return (
      <div className="main-container" id="new-class-main-container">
        <div className="new-class-header">
          <h1>New FAQ</h1>
        </div>
        <div className="new-class-form-container">
          <form  onSubmit={this.onFormSubmit} className="new-class-form">
            <label htmlFor="question"> Question:</label>
            <textarea rows="10" cols="50" value={this.state.question} onChange={this.onInputChange} name="question" id="description"/>
            <label htmlFor="answer"> Answer:</label>
            <textarea rows="10" cols="50" value={this.state.answer} onChange={this.onInputChange} name="answer" id="description"/>
            <div className="classes-buttons">
              <button onClick={this.onFormSubmit}>Submit</button>
              <Link to="/faqs"><button>Back</button></Link>
            </div>
          </form>
        </div>
      </div>
    )
  }
}

export default withRouter(NewFaq);