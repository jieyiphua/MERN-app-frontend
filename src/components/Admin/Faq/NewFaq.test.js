import NewFaq from "./NewFaq";
import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";
import sinon from "sinon";

describe("<NewFaq />", () => {
  it("renders component", () => {
    const wrapper = shallow(<NewFaq />);
    expect(wrapper).toMatchSnapshot();
  });
});
