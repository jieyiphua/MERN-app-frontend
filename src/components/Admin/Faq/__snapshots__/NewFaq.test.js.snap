// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewFaq /> renders component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
